txnsync/txngroups.go

```diff
@@ -60,10 +99,46 @@ func decodeTransactionGroups(bytes []byte) (txnGroups []transactions.SignedTxGro
 	return txnGroups, nil
 }
 
+func decodeTransactionGroups(bytes []byte, genesisID string, genesisHash crypto.Digest) (txnGroups []transactions.SignedTxGroup, err error) {
+	if len(bytes) == 0 {
+		return nil, nil
+	}
+	var stub txGroupsEncodingStub
+	_, err = stub.UnmarshalMsg(bytes)
+	if err != nil {
+		return nil, err
+	}
+
+	stx := make([]transactions.SignedTxn, stub.TotalTransactionsCount)
```

Comment:
add a test here to ensure `stub.TotalTransactionsCount` isn't too large. We don't want someone to send us 10G for this field, and we will crash as a result.